elephant/signal_processing.py

```diff
@@ -136,6 +137,129 @@ def zscore(signal, inplace=True):
+        dt = 0.02
+        N = 2018
+        f = 0.5
+        t = np.arange(N)*dt
+        x = np.zeros((N,2))
+        x[:,0] = 0.2 * np.sin(2.*np.pi*f*t)
+        x[:,1] = 5.3 * np.cos(2.*np.pi*f*t)
+        # Generate neo.AnalogSignals from x
+        signal = neo.AnalogSignal(x, units='mV', t_start=0.*pq.ms,
+            sampling_rate=1/dt*pq.Hz, dtype=float)
+        rho = elephant.signal_processing.cross_correlation_function(
+            signal, [0,1], nlags=150)
+        env = elephant.signal_processing.cross_correlation_function(
+            signal, [0,1], nlags=150, env=True)
+        plt.plot(rho.times, rho)
+        plt.plot(env.times, env) # should be equal to one
+        plt.show()
+    """
+
+    # Make ch_pairs a 2D array
+    pairs = np.array(ch_pairs)
+    if pairs.ndim == 1:
+        pairs = pairs[:, np.newaxis]
+
+    # Check input
+    if not isinstance(signal, neo.AnalogSignal):
+        raise ValueError('Input signal is not a neo.AnalogSignal!')
+    if np.shape(pairs)[1] != 2:
+        pairs = pairs.T
+    if np.shape(pairs)[1] != 2:
+        raise ValueError('ch_pairs is not a list of channel pair indices.'\
+                         'Cannot define pairs for cross-correlation.')
+    if not isinstance(env, bool):
+        raise KeyError('env is not a boolean!')
+    if nlags is not None:
+        if not isinstance(nlags, (int, float)):
+            raise KeyError('nlags must be an integer or float larger than 0!')
+        if nlags <= 0:
+            raise KeyError('nlags must be an integer or float larger than 0!')
+
+    # z-score analog signal and store channel time series in different arrays
+    # Cross-correlation will be calculated between xsig and ysig
+    xsig = np.array([zscore(signal).magnitude[:, pair[0]] \
+        for pair in pairs]).T
+    ysig = np.array([zscore(signal).magnitude[:, pair[1]] \
+        for pair in pairs]).T
+
+    # Define vector of lags tau
+    nt, nch = np.shape(xsig)
+    tau = (np.arange(nt) - nt//2)
+
+    # Calculate cross-correlation by taking Fourier transform of signal,
+    # multiply in Fourier space, and transform back. Correct for bias due
+    # to zero-padding
+    xcorr = np.zeros((nt, nch))
+    for i in range(nch):
+        xcorr[:, i] = scipy.signal.fftconvolve(xsig[:, i], ysig[::-1, i],
+                                               mode='same')
+    xcorr = xcorr / npm.repmat((nt-abs(tau)), nch, 1).T
+
+    # Calculate envelope of cross-correlation function with Hilbert transform.
+    # This is useful for transient oscillatory signals.
+    if env:
+        for i in range(nch):
+            xcorr[:, i] = np.abs(scipy.signal.hilbert(xcorr[:, i]))
+
+    # Cut off lags outside desired range
+    if nlags is not None:
+        nlags = int(np.round(nlags))
+        tau0 = int(np.argwhere(tau == 0))
+        xcorr = xcorr[tau0-nlags:tau0+nlags+1, :]
+
+    # Return neo.AnalogSignal
+    cross_corr = neo.AnalogSignal(xcorr,
+                                  units='',
```

Comment:
Why do you give an empty string for the `units`

Reply:
correlation is unitless

Reply:
Right, didn't even now that this is possible in `neo`.